File: Count the Number of Consistent Strings.py
class Solution:
    def countConsistentStrings(self, allowed: str, words: List[str]) -> int:
        allowed = set(allowed)
        count = 0
        for word in words:
            can = True
            for c in word:
                if c not in allowed:
                    can = False
                    break
            count += int(can)
        return count
    

File: Count the Number of Consistent Strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        unordered_set<char> seen(allowed.begin(), allowed.end());
        int count = 0;
        for (auto word: words) {
            bool can = true;
            for (char c: word) {
                if (!seen.contains(c)) {
                    can = false;
                    break;
                }
            }
            if (can) count++;
        }
        return count;
    }
};
